graph/tarjan.cpp

```diff
@@ -0,0 +1,236 @@
+/**
+ * @file
+ * @brief Implementation for [Tarjan's articulation points algorithm](https://en.wikipedia.org/wiki/Biconnected_component#Algorithms)
+ * @author [Nisarg Pipaliya](https://github.com/NisargPipaliya)
+ * @details From Wikipedia:
+ * The classic sequential algorithm for computing biconnected components
+ * in a connected undirected graph is due to John Hopcroft and Robert Tarjan (1973).
+ * It runs in linear time, and is based on depth-first search.
+ * The idea is to run a depth-first search while maintaining the following information:
+ *      1.the depth of each vertex in the depth-first-search tree (once it gets visited), and
+ *      2.for each vertex v, the lowest depth of neighbors of all descendants of v (including v itself) in the depth-first-search tree, called the lowpoint.
+ */
+/**
+ * @brief Find articulation points with Tarjan's algorithm
+ * @param tg Pointer to target graph
+ * @param start_vertex Location to start the search
+ * @param result Pointer to location to store the results
+ */
+ 
+#include<iostream> /// for general input and output
+#include<vector> /// for storing graph
+#include<queue> /// used for BFS algorithm in below namespace
+#include<list>  /// used for making adjacency list representation of graph
+#include<set>   /// for returning result of tarjan's agorithm
+
+/**
+ * @namespace graph
+ * @brief Graph algorithms
+ */
+namespace graph {
+class Graph
+{
+public:
+
+    std::vector<std::list<std::pair<int, int>>> g;   // adjecency list to store graph
+    int vertex;                                     // Total number of vertices in graph
```

Comment:
`///<` should be used so that Doxygen can detect these variables properly.
```suggestion
    std::vector<std::list<std::pair<int, int>>> g;   ///< adjecency list to store graph
    int vertex;                                     ///< Total number of vertices in graph
```